app/controllers/events_controller.rb

```diff
@@ -0,0 +1,73 @@
+class EventsController < ApplicationController
+  before_action :set_project
```

Comment:
I haven't used Devise gem before but here's what I found: https://github.com/plataformatec/devise/wiki/How-To:-Define-resource-actions-that-require-authentication-using-routes.rb
I thought it's nice to keep user authorization logic away from the controller but maybe there are some better conventions I don't know about.